### PR TITLE
bpo-43926: Cleaner metadata with PEP 566 JSON support.

### PR DESCRIPTION
**Pull request by jaraco**
https://bugs.python.org/issue43926

### CODE REVIEW

**Comment by jaraco** [comment]
Note, this change includes and should be merged after #25297.

File: Doc/library/importlib.metadata.rst [hunk 1]
```diff
@@ -170,6 +170,12 @@ the values are returned unparsed from the distribution metadata::
     >>> wheel_metadata['Requires-Python']  # doctest: +SKIP
     '>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*'
 
+``PackageMetadata`` also presents a ``json`` attribute that returns
```
Comment (tirkarthi) [inline Doc/library/importlib.metadata.rst:173]: This is a Python 3.10 only feature and hence a versionadded directive can be added.

Slightly out of topic, are `Distribution` objects and `metadata` attribute immutable? Accessing json attribute from metadata objects repeats the computation though it looks like a computed once attribute so just curious if `metadata` attribute of Distribution, `json` attribute etc. can be made a cached_property.

Comment (jaraco) [reply Doc/library/importlib.metadata.rst:173]: Fixed in 289665029e (also `versionchanged` for changes to the interpretation of the metadata).

**Comment by jaraco** [comment]
@warsaw I note also that while we intentionally consolidated `importlib.metadata` into a single module (#15153), but the package has since diversified and the `_common` module now conflicts with `importlib_resources`, so I've moved the modules back into a package to avoid the conflict and more clearly separate the concerns.

File: Doc/library/importlib.metadata.rst (hunk 1 above)
Comment (jaraco) [reply Doc/library/importlib.metadata.rst:173]: > can be made a cached_property?

Although technically the PackageMetadata object is mutable, the protocol for it doesn't allow for mutation, so it's effectively immutable, so yes, a `cached_property` could be employed. Such an optimization may be premature, so I'm slightly inclined to defer that concern until a use-case presents itself. I don't feel strongly about it though and would likely accept a PR to `importlib_metadata`.

**Comment by jaraco** [comment]
In order to ensure this change is included in the 3.10 release, I'm merging now. Happy to troubleshoot or back out specific behaviors as needed.

**Pull request merged**